src/main/java/uk/ac/ic/wlgitbridge/bridge/Bridge.java

```diff
@@ -174,9 +177,21 @@ public static Bridge make(
             SwapStore swapStore,
             SnapshotApi snapshotApi
     ) {
-        ProjectLock lock = new ProjectLockImpl((int threads) ->
-                Log.info("Waiting for " + threads + " projects...")
-        );
+        ProjectLock lock;
```

Comment:
Oooh, I *almost* want a `ProjectLockFactory`. Almost. This is actually fine. :-)